# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptSettings.h
# Commit: 78181c32bdc6
# Full Hash: 78181c32bdc6b1d66eac04f3ce11a2902f477823
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-27 21:52:52
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r 912cf1adac80 -r 78181c32bdc6 dom/script/ScriptSettings.h
--- a/dom/script/ScriptSettings.h	Thu Feb 27 16:17:34 2025 +0000
+++ b/dom/script/ScriptSettings.h	Thu Feb 27 16:17:35 2025 +0000
@@ -35,6 +35,7 @@
 namespace dom {
 
 class Document;
+class WebTaskSchedulingState;
 
 /*
  * Per thread setup/teardown routines. Init and Destroy should be invoked
@@ -95,6 +96,8 @@
 // Returns the global associated with the current compartment. This may be null.
 nsIGlobalObject* GetCurrentGlobal();
 
+WebTaskSchedulingState* GetWebTaskSchedulingState();
+
 // JS-implemented WebIDL presents an interesting situation with respect to the
 // subject principal. A regular C++-implemented API can simply examine the
 // compartment of the most-recently-executed script, and use that to infer the